circuit_knitting/cutting/cut_finding/best_first_search.py

```diff
@@ -0,0 +1,386 @@
+# This code is a Qiskit project.
+
+# (C) Copyright IBM 2024.
+
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Classes required to implement Dijkstra's (best-first) search algorithm."""
+
+from __future__ import annotations
+
+import heapq
+import numpy as np
+from numpy.typing import NDArray
+from numpy.random._generator import Generator
```

Comment:
Interesting import of `_generator` here. Haven't seen this before. Is this stable?

Reply:
Good catch, changing this to `from numpy.random import Generator`

Reply:
'Tis done: [002144e](https://github.com/Qiskit-Extensions/circuit-knitting-toolbox/pull/520/commits/002144eab5a2b2a5a5304226ca6a084e21db8c7c)